packages/webapp/src/containers/InviteSignUp/index.jsx

```diff
@@ -68,37 +73,76 @@ function InviteSignUp({ history }) {
       });
     }
   };
-  const clientId = import.meta.env.VITE_GOOGLE_OAUTH_CLIENT_ID;
+
+  const login = useGoogleLogin({
+    onSuccess: async (tokenResponse) => {
+      try {
+        // I have not previously used redux or redux-saga, but I imagine a query to an API like this should be in a saga.js file?
+        const userInfo = await axios.get('https://www.googleapis.com/oauth2/v3/userinfo', {
+          headers: { Authorization: `Bearer ${tokenResponse.access_token}` },
+        });
+        onSuccessGoogle(userInfo.data, tokenResponse);
+      } catch (error) {
+        console.log(error);
+      }
+    },
+    onError: (errorResponse) => console.log(errorResponse),
+  });
+
+  /* Shape of response object from hook:
+
+      { 
+        access_token: ...,
+        authuser: "2",
+        expires_in: 3599
+        prompt: "none",
+        scope: "email profile ...",
+        token_type: "Bearer"
+      }
+
+
+    Shape of response from https://www.googleapis.com/oauth2/v3/userinfo using the access_token above:
+
+      { 
+        email: ...
+        email_verified: true,
+        family_name: ...
+        given_name: ...
+        locale: ...,
+        name: ...
+        picture: <link>
+        sub: ...
+      }
+
+  */
+
+  /* ProceedButton is a wrapper for the renderProps that handles the onClick similar to how the previous package's GoogleLogin component did: https://github.com/anthonyjgrove/react-google-login/blob/master/src/google-login.js I wrote it like this in order to keep the surrounding code unchanged. I don't really use/understand renderProps myself so I will defer to someone else if maybe they can be refactored out, now that our new package doesn't require them. */
```

Comment:
@kathyavini 
I too am not familiar with this syntax. How it is right now it doesn't look to me like it fits the pattern I have seen in the React docs. 

From what I can tell the previous GoogleLogin returned a component, whereas the new google login is a hook. I am not confident keeping this syntax is appropriate.

Reply:
Yeah, I have also never seen or used a pattern like this before. I kept the render props because I think the next component up the tree (that imports this one) expects them. But I'm definitely open to a refactor; I was hoping that someone more familiar with this structure might chime in on the simplest way to do that!